src/HotChocolate/Core/src/Types.OffsetPagination/CollectionSegment~1.cs

```diff
@@ -0,0 +1,38 @@
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace HotChocolate.Types.Pagination
+{
+    /// <inheritdoc />
+    public class CollectionSegment<T> : CollectionSegment
+        where T : class
```

Comment:
@michaelstaib Is this fine? i didn't find a quicker solution for being able to pass it to the base constructor which needs an `IReadOnlyList<object>`.

Reply:
This aligns great with the connection types.